src/wallet/wallet_rpc_payments.cpp

```diff
@@ -146,7 +146,14 @@ bool wallet2::search_for_rpc_payment(uint64_t credits_target, const std::functio
     if (major_version >= RX_BLOCK_VERSION)
     {
       const int miners = 1;
-      crypto::rx_slow_hash(height, seed_height, seed_hash.data, hashing_blob.data(), hashing_blob.size(), hash.data, miners, 0);
+      try { crypto::rx_slow_hash(height, seed_height, seed_hash.data, hashing_blob.data(), hashing_blob.size(), hash.data, miners, 0); }
+      catch (const std::exception &e)
+      {
+        MERROR("Hashing error");
+        if (errorfunc)
+          errorfunc("Hashing error");
+        return false;;
```

Comment:
One of the possible exceptions is caused from systemd rejecting W^X pages (the default, for performance reasons) when configured to do so. In this case, it is not recoverable. If the issue is transient, the code allows it to recover. This was synthetic though (something like if (rand()%256==0) throw). Since we now know systemd is the reason, I think this can be dropped tbh.

Reply:
I think you mean systemd is rejecting W|X pages?

Reply:
It turns out it's rejecting X on pages previously W, even if it's not at the same time, when a setting is set. And people tend to set this setting, because it's name hints it's only about W^X.